lib/fencing.py.py

```diff
@@ -33,6 +34,13 @@
 
 LOG_FORMAT = "%(asctime)-15s %(levelname)s: %(message)s"
 
+OPTIONS_TYPE = Dict[str, Any]
+SET_POWER_FN_TYPE = Callable[[Optional["fspawn"], OPTIONS_TYPE], Optional[NoReturn]]
```

Comment:
Typing is new in 3.5, so we should not use it in the fencing library, as it needs to be backwards compatible Python 2.7. Ideally we should keep new agents compatible with most 3.x versions as well.

https://docs.python.org/3/library/typing.html

Reply:
If I would have known that Python 2.7 compatibility is required, I would have not done these changes of course. When do you plan on dropping Python 2.7 & <Python 3.6 support? I would like to open a PR with typing hints again. It took me quite some hours to reverse-engineer `fencing.py.py` to a degree that I was able to code the fence to my satisfaction.

Reply:
I think you should be able to do most of this without the changes to the library.

fence_ilo_ssh and fence_cyberpower_ssh should give some good pointers.

Reply:
Yeah, exactly my point. They give me pointers to reverse engineer but I would not expect to reverse engineer a project to contribute to it. Anyway, I will drop the typing in a minute and push again.